webgui/scripts/codecompass/model.js

```diff
@@ -1,60 +1,59 @@
 define([
-  'codecompass/urlHandler'],
-function (urlHandler) {
-
-  //--- Model object ---//
-
-  var model = {
-    _fileTypesToService : {},
-
-    /**
-     * Get service identified by name. Registrate if doesn't exist.
-     */
-    addService : function (name, url, Client) {
-      if (!this[name]) {
-        // These two services are independent from the workspaces, so these are
-        // on an URL which don't contain the workspace name.
-        if (url !== 'PluginService' && url !== 'WorkspaceService')
-          url = workspace + '/' + url;
-
-        var service
-          = new Client(new Thrift.Protocol(new Thrift.Transport(url)));
-
-        //--- Map filetypes to service for each language service ---//
-
-        if (service.getFileTypes){
-          service.getFileTypes().forEach(function (fileType) {
-            model._fileTypesToService[fileType] = service;
-          });
-        }
-
-        this[name] = service;
+  'codecompass/urlHandler',
+  'exports'],
```

Comment:
For full information see this page: https://dojotoolkit.org/documentation/tutorials/1.9/modules_advanced/

E.g.:
```js
module A depends on module B
module B depends on module A

In main we include module A.

- moduleA is temporarily resolved as a reference to an empty object.
- moduleB is resolved.
- moduleA is fully resolved.
```